redash/query_runner/athena.py

```diff
@@ -118,13 +140,31 @@ def annotate_query(cls):
     def type(cls):
         return "athena"
 
-    def __get_schema_from_glue(self):
-        client = boto3.client(
-                'glue',
-                aws_access_key_id=self.configuration.get('aws_access_key', None),
-                aws_secret_access_key=self.configuration.get('aws_secret_key', None),
-                region_name=self.configuration['region']
+    def _get_iam_credentials(self, user=None):
+        if ASSUME_ROLE:
+            role_session_name = 'redash' if user is None else str(user.email)
+            sts = boto3.client('sts')
+            creds = sts.assume_role(
+                RoleArn=self.configuration.get('iam_role'),
+                RoleSessionName=role_session_name,
+                ExternalId=self.configuration.get('external_id')
                 )
+            return {
+                'aws_access_key_id': creds['Credentials']['AccessKeyId'],
+                'aws_secret_access_key': creds['Credentials']['SecretAccessKey'],
+                'aws_session_token': creds['Credentials']['SessionToken'],
+                'region_name': self.configuration['region']
+            }
+        else:
+            return {
+                'aws_access_key_id': self.configuration.get('aws_access_key', None),
+                'aws_secret_access_key': self.configuration.get('aws_secret_key', None),
+                'aws_session_token': None,
```

Comment:
Considering we weren't passing this before, maybe it's safer to omit this option instead of passing `None`?

```suggestion
```

Reply:
yep, fair point, removed.